lib/Sema/MiscDiagnostics.cpp

```diff
@@ -3194,6 +3194,20 @@ performTopLevelDeclDiagnostics(TopLevelCodeDecl *TLCD) {
   TLCD->walk(checker);
 }
 
+/// Perform diagnostics on closure body.
+void swift::performClosureBodyDiagnostics(ClosureExpr *closure) {
+  if (auto *ACE = dyn_cast<AbstractClosureExpr>(closure)) {
+    // Skip if we aleady checked variable usage.
```

Comment:
Can we instead change those walkers to skip closure bodies, so that we can check them unconditionally here?

Reply:
The walkers can't skip the closure bodies entirely because they look for uses of a variable within the closure. But, I could try to skip the tracking of any variables declared within the closures (I'm not sure yet how trivial this is, but I will look into it). Then later, walk through all closures and track variable declarations unconditionally.

Thank you for the feedback!

Reply:
You can’t skip them in the walker itself, but you can skip them when you first enter the walker. E.g. instead of walking the entire Top Level Code Decl, you can iterate over its members and skip closures knowing they’ll be checked later.